app/src/main/java/com/gusta/wakemehome/utilities/PermissionUtils.java

```diff
@@ -0,0 +1,117 @@
+
+import com.gusta.wakemehome.BuildConfig;
+import com.gusta.wakemehome.R;
+import com.gusta.wakemehome.geofencing.GeofenceManager;
+
+public final class PermissionUtils {
+
+    // Constant for logging
+    private static final String TAG = GeofenceManager.class.getSimpleName();
+
+    /**
+     * Return the current state of the permissions needed.
+     */
+    public static boolean missingPermissions(Context context) {
+        int permissionState = ActivityCompat.checkSelfPermission(context,
+                Manifest.permission.ACCESS_FINE_LOCATION);
+        return permissionState != PackageManager.PERMISSION_GRANTED;
+    }
+
+    /**
+     * Ask the user for permission.
+     */
+    public static void requestPermissions(final Activity activity) {
+
+        // Request permissions from the user
+        boolean shouldProvideRationale =
+                ActivityCompat.shouldShowRequestPermissionRationale(activity,
+                        Manifest.permission.ACCESS_FINE_LOCATION);
+
+        // Provide an additional rationale to the user. This would happen if the user denied the
+        // request previously, but didn't check the "Don't ask again" checkbox.
+        if (shouldProvideRationale) {
+            Log.i(TAG, "Displaying permission rationale to provide additional context.");
+            NotificationUtils.notifyUser(activity, R.string.permission_rationale,
+                    android.R.string.ok, new View.OnClickListener() {
+                        @Override
+                        public void onClick(View view) {
+                            // Request permission
+                            ActivityCompat.requestPermissions(activity,
+                                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
+                                    Constants.ACCESS_FINE_LOCATION_PERMISSION_REQUEST_CODE);
+                        }
+                    });
+        } else {
+            Log.i(TAG, "Requesting permission");
+            // Request permission. It's possible this can be auto answered if device policy
+            // sets the permission in a given state or the user denied the permission
+            // previously and checked "Never ask again".
+            ActivityCompat.requestPermissions(activity,
+                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
+                    Constants.ACCESS_FINE_LOCATION_PERMISSION_REQUEST_CODE);
+        }
+    }
+
+    /**
+     * Callback received when a permissions request has been completed. Redirected from activity.
+     */
+    public static void handleResult(final Activity activity, PendingTaskHandler pendingTaskHandler,
+                                    @NonNull int[] grantResults) {
+        if (grantResults.length <= 0) {
+            // If user interaction was interrupted, the permission request is cancelled and you
+            // receive empty arrays.
+            Log.i(TAG, "User interaction was cancelled.");
```

Comment:
All the string into `Strings.xml`, it should be able to be translatable

Reply:
Debug log.
No need to translate log. Only things that user sees.